### ISSUE TITLE
When using FreeTDS .95 autocommit becomes enabled

### ISSUE THREAD

**Issue by Asara** [FreeTDS/freetds#50]
Currently using pyodbc to make transactions.

Using FreeTDS .91-2, pyodbc would set autocommit to off, and the driver would respect it.

Seemingly, under FreeTDS 0.95.19 (shipped with CentOS 7) , once an initial commit is made, all further transactions are commit automatically.  pyodbc shows autocommit turned off, but everything is commit by itself.

We are using the same exact codebase with the same version of pyodbc/python.

Any help would be appreciated.  Thanks!

**Comment by freddy77** [comment]
Can you send a TDSDUMP and a ODBC trace ?

**Comment by Asara** [comment]
We have been doing some debugging and came to some realizations.

It seems as if the autocommit off is implemented differently between the two version.

For version .91:

    write.c:140:tds_put_string converting 28 bytes of "SET IMPLICIT_TRANSACTIONS ON"

Whereas for version .95:

    mem.c:648:tds_free_all_results()
    util.c:165:Changed query state from IDLE to WRITING
    util.c:165:Changed query state from WRITING to PENDING
    packet.c:740:Sending packet
    0000 01 01 00 2a 00 00 01 00-42 00 45 00 47 00 49 00 |...*.... B.E.G.I.|
    0010 4e 00 20 00 54 00 52 00-41 00 4e 00 53 00 41 00 |N. .T.R. A.N.S.A.|
    0020 43 00 54 00 49 00 4f 00-4e 00                   |C.T.I.O. N.|


This is affecting the second transaction, because when you commit a transaction, it doesn't open a new transaction, and as such gives us:

    error.c:565:SQLGetDiagRec: "[FreeTDS][SQL Server]The COMMIT TRANSACTION request has no corresponding BEGIN TRANSACTION."


Thanks!

**Comment by Asara** [comment]
After some more investigation, we saw that the implementation of committing was changed.  If a commit is initiated via pyodbc, then there is a begin tran after the commit, but because we changed the implementation of commit in our code, we only send a commit and not a commit - begin tran.  

Is there a way to use SET IMPLICIT_TRANSACTION ON instead of commit/begin tran, since it seems to be easier to manage from a DBA perspective.  

Thank you!
